### PR TITLE
[python] Do not use extent 2048 on 8-bit index types

### LINKED ISSUE #652: [python] Unpin TileDB-Py 0.19.0
Reminder to revert https://github.com/single-cell-data/TileDB-SOMA/pull/651 after analysis

### PR DESCRIPTION
**Pull request by johnkerl**
Confluence of several issues today:

* Old bug described by #518, fixed on this PR -- was not fixed until just now
  * Root cause was the 2048 part fixed on this PR -- which mods to 0 of course for 8-bit types
  * Constructing a tiledb dim with extent 0 _should_ fail -- and was failing in the xfailed cases on this PR
* TileDB-Py 0.19.1 included a (completely innocent) refactor in terms of how `tiledb.Dimension` is constructed and handled
  * Giving rise to https://github.com/pytest-dev/pytest/issues/4659
* So in summary (debug thanks to @nguyenv) we had soma extent 0 -> tiledb-py constructor fail as designed-> segv in unit test due to pytest bug -> which is what we were seeing today with #651

### CODE REVIEW

File: apis/python/src/tiledbsoma/dataframe.py [hunk 1]
```diff
@@ -100,10 +100,19 @@ def _create_empty(
                 else:
                     raise TypeError(f"Unsupported dtype {dtype}")
 
+            extent = create_options.dim_tile(index_column_name)
+            if (
+                dtype == pa.int8()
+                or dtype == pa.uint8()
+                or dtype == np.int8
+                or dtype == np.uint8
+            ):
```
Comment (gsakkis) [inline apis/python/src/tiledbsoma/dataframe.py:109]: Minor drive-by question: can the two first conditions happen? `dtype` here is the result of `util_arrow.tiledb_type_from_arrow_type`, which IIUC can only be a `np.dtype` instance or the string "ascii" or "bytes". So this condition can be simplified to `if isinstance(dtype, np.dtype) and dtype.itemsize == 1:` 
(or perhaps more tersely `if getattr(dtype, "itemsize", 0) == 1:`).

**Review by mlin — COMMENTED** [review]
@johnkerl 

1. Should we put a comment inline explaining the hard-coded `extent` value?
2. How do these issues affect the new PyPI package (and upcoming conda package), which I believe still require tiledb `>=0.19.0`?

**Comment by johnkerl** [comment]
@mlin 

> Should we put a comment inline explaining the hard-coded extent value?

Great idea, thanks!

> How do these issues affect the new PyPI package (and upcoming conda package), which I believe still require tiledb >=0.19.0?

I just changed it to `>=0.19.0`, thanks!

File: apis/python/src/tiledbsoma/dataframe.py (hunk 1 above)
Comment (johnkerl) [reply apis/python/src/tiledbsoma/dataframe.py]: Awesome, thanks @gsakkis !

**Review by gsakkis — APPROVED** [review]

File: apis/python/setup.cfg [hunk 2]
```diff
@@ -36,7 +36,7 @@ install_requires =
     pyarrow
     scanpy
     scipy
-    tiledb==0.19.0
+    tiledb>=0.19.0
     typing_extensions
```
Comment (bkmartinjr) [inline apis/python/setup.cfg:40]: I believe the correct spelling is `typing-extensions`.

See https://pypi.org/project/typing-extensions/

yes, the package name and import name are different!

**Review by bkmartinjr — COMMENTED** [review]
Apologies for the late review - would like to see the issues fixed (but another PR is fine).

File: apis/python/setup.cfg (hunk 2 above)
Comment (johnkerl) [reply apis/python/setup.cfg:40]: @bkmartinjr #655

File: apis/python/src/tiledbsoma/dataframe.py [hunk 3]
```diff
@@ -100,10 +100,15 @@ def _create_empty(
                 else:
                     raise TypeError(f"Unsupported dtype {dtype}")
 
+            # Default 2048 mods to 0 for 8-bit types and 0 is an invalid extent
+            extent = create_options.dim_tile(index_column_name)
```
[the remark this answers is not in the document]
Comment (johnkerl) [reply apis/python/src/tiledbsoma/dataframe.py:104]: @bkmartinjr no because the two other callsites both have hard-coded `dtype` of `np.int64`:

* https://github.com/single-cell-data/TileDB-SOMA/blob/main/apis/python/src/tiledbsoma/sparse_nd_array.py#L82
* https://github.com/single-cell-data/TileDB-SOMA/blob/main/apis/python/src/tiledbsoma/dense_nd_array.py#L82
